### ISSUE TITLE
Get Clasp to always exit using Ctrl-D

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#105]
Sometimes Clasp ignores Ctrl-D and just keeps prompting rather than exiting.

**Comment by Shinmera** [comment]
"Sometimes"? Under which circumstances?

**Comment by drmeister** [comment]
The low-level debugger should exit now on Ctrl-D

**Comment by Shinmera** [comment]
I don't know if this is the right way to confirm this, but on fb10987:
```
> (+1 4)               
terminate called after throwing an instance of 'std::invalid_argument'
  what():  mpz_set_str
../../src/gctools/memoryManagement.cc:72 Trapped SIGABRT - starting debugger
ABORT was called!!!!!!!!!!!!

"# 60/b@0x7f6710336b49         readtable.cc  1236/0   (CORE:READER-LIST-ALLOW-CONSING-DOT #<COMMON-LISP:SYNONYM-STREAM @0x7f67086f3e68) >  #\\( )
" 
"The following restarts are available:" 
"ABORT      a    Abort to REPL" 
Frame-60-Dbg[2]>terminate called recursively
../../src/gctools/memoryManagement.cc:72 Trapped SIGABRT - starting debugger
ABORT was called!!!!!!!!!!!!
```
etc

**Comment by Shinmera** [comment]
Still there on 482d6ca5bd8ad30ff8c9b6a835989f3aec395e1a

**Comment by Shinmera** [comment]
I think this is fixed by now, but @drmeister is there any way to drop into the low-level debugger explicitly?

**Comment by drmeister** [comment]
There isn't any way to drop into the low-level debugger from cclasp. 
From aclasp it is the only debugger:

```bash-3.2$ cd /Volumes/spin/Development/clasp/build/boehmdc
bash-3.2$ ls
Contents			cclasp-boehmdc			compile_commands.json		iclasp-boehmdc.dSYM
aclasp-boehmdc-common-lisp.bc	cclasp-boehmdc-common-lisp.bc	config.h			iclasp-boehmdc.lto.o
aclasp-boehmdc-image.fasl	cclasp-boehmdc-image.fasl	fasl				src
bclasp-boehmdc-common-lisp.bc	cclasp-boehmdc.dSYM		generated
bclasp-boehmdc-image.fasl	cclasp-boehmdc.lto.o		iclasp-boehmdc

bash-3.2$ ./iclasp-boehmdc -i aclasp-boehmdc-image.fasl 

CORE> (error "foo")
foo

universalErrorHandler
../../src/core/foundation.cc:263 dbg_hook(...) was called

"# 17/b              lisp.cc  2006 (UNIVERSAL-ERROR-HANDLER NIL \"foo\" NIL )
" 
"The following restarts are available:" 
"ABORT      a    Abort to REPL" 
"Frame-17-Dbg[1]> " ../../src/core/debugger.cc:117 Exiting debugger

bash-3.2$ ```

**Comment by Shinmera** [comment]
Can confirm fixed on 99669d674a6f2b7ce0a9127cf6537fc03066baa5
```
./iclasp-boehmdc -i aclasp-boehmdc-image.fasl

CORE> (error "hi")
hi

universalErrorHandler
../../src/core/foundation.cc:263 dbg_hook(...) was called

"# 17/b              lisp.cc  2006 (UNIVERSAL-ERROR-HANDLER NIL \"hi\" NIL )
" 
"The following restarts are available:" 
"ABORT      a    Abort to REPL" 
"Frame-17-Dbg[1]> " ../../src/core/debugger.cc:117 Exiting debugger
```
